pkg/proxy/proxy.go

```diff
@@ -220,6 +220,62 @@ func fillIdentity(info *accesslog.EndpointInfo, id policy.NumericIdentity) {
 	}
 }
 
+// fillEndpointInfo tries to resolve the IP address and fills the EndpointInfo
+// fields with either ReservedIdentityHost or ReservedIdentityWorld
+func fillEndpointInfo(info *accesslog.EndpointInfo, ip net.IP) {
```

Comment:
This function was really difficult to follow, especially the implicit "default" case when an endpoint is none of World, Cluster or Host! Since we're just moving existing code, it's probably not in-scope to refactor this but could you add more explanation of which conditions result in which outcome. Either in the function description, which needs to say it also returns ReservedIdentityCluster and a unreserved Endpoint's info, or inline comments to make it more clear how/why the code ends up and when it exits (the returns are clear, the case without the return I barely noticed!).
If you prefer to refactor that's also fine! I just didn't want to bloat this PR.

Reply:
Added more comments to make the flow obvious.